inc/class-wpseo-admin-bar-menu.php

```diff
@@ -470,61 +511,56 @@ protected function get_title() {
 	/**
 	 * Gets the current post if in a singular post context.
 	 *
-	 * @global string       $pagenow Current page identifier.
+	 * @return bool|int False when not singular, Post ID when it is.
+	 *
 	 * @global WP_Post|null $post    Current post object, or null if none available.
 	 *
-	 * @return WP_Post|null Post object, or null if not in singular context.
+	 * @global string       $pagenow Current page identifier.
 	 */
-	protected function get_singular_post() {
+	protected function get_singular_post_id() {
 		global $pagenow, $post;
 
 		if ( ! is_singular() && ( ! is_blog_admin() || ! WPSEO_Metabox::is_post_edit( $pagenow ) ) ) {
-			return null;
+			return false;
 		}
 
 		if ( ! isset( $post ) || ! is_object( $post ) || ! $post instanceof WP_Post ) {
-			return null;
+			return false;
 		}
 
-		return $post;
+		return $post->ID;
 	}
 
 	/**
 	 * Gets the focus keyword for a given post.
 	 *
-	 * @param WP_Post $post Post object to get its focus keyword.
+	 * @param int $post_id Post ID.
 	 *
-	 * @return string Focus keyword, or empty string if none available.
+	 * @return string|bool Focus keyword, or empty string if none available. False when we shouldn't have one.
 	 */
-	protected function get_post_focus_keyword( $post ) {
-		if ( ! is_object( $post ) || ! property_exists( $post, 'ID' ) ) {
-			return '';
-		}
-
+	protected function get_post_focus_keyword( $post_id ) {
 		/**
 		 * Filter: 'wpseo_use_page_analysis' Determines if the analysis should be enabled.
 		 *
 		 * @api bool Determines if the analysis should be enabled.
 		 */
 		if ( apply_filters( 'wpseo_use_page_analysis', true ) !== true ) {
-			return '';
+			return false;
 		}
 
-		return WPSEO_Meta::get_value( 'focuskw', $post->ID );
+		$focus_keyword = WPSEO_Meta::get_value( 'focuskw', $post_id );
+
+		return $focus_keyword;
```

Comment:
Couldn't `WPSEO_Meta::get_value( 'focuskw', $post_id )` just be returned without declaring and assigning a new variable `$focus_keyword`?

Reply:
Yes